sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/InferSchema.scala

```diff
@@ -30,22 +30,37 @@ import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.UTF8String
 
-private[csv] object CSVInferSchema {
+private[csv] object InferSchema {
 
   /**
    * Similar to the JSON schema inference
    *     1. Infer type of each row
    *     2. Merge row types to find common type
    *     3. Replace any null types with string type
    */
-  def infer(
-      tokenRdd: RDD[Array[String]],
-      header: Array[String],
-      nullValue: String = ""): StructType = {
+  def infer(csv: RDD[String], options: CSVOptions): StructType = {
```

Comment:
This looks very similar to `DefaultSource.inferSchema` why not move the common functionality  into a single method?